sql/core/src/test/scala/org/apache/spark/sql/StringFunctionsSuite.scala

```diff
@@ -663,4 +664,18 @@ class StringFunctionsSuite extends QueryTest with SharedSparkSession {
         start = 7,
         stop = 47))
   }
+
+  test("SPARK-41780: INVALID_PARAMETER_VALUE - invalid parameters `regexp` in regexp_replace") {
```

Comment:
Just found one more test maybe related to this changes:
https://github.com/apache/spark/blob/d0a598922e9df8ea90125d22c241810feab06017/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/RegexpExpressionsSuite.scala#L519-L536

Maybe can we address this together in this work as well ??

Reply:
Ok, Let me use `checkErrorInExpression` replace `checkExceptionInExpression ` to check exception.